okdownload/src/main/java/com/liulishuo/okdownload/core/breakpoint/BreakpointInfo.java

```diff
@@ -124,12 +124,13 @@ public void setEtag(String etag) {
 
     public long getTotalOffset() {
         long offset = 0;
-        ArrayList<BlockInfo> list = (ArrayList<BlockInfo>) ((ArrayList) blockInfoList).clone();
-
-        final int count = list.size();
-        for (int i = 0; i < count; i++) {
-            final BlockInfo info = list.get(i);
-            offset += info.getCurrentOffset();
+        final Object[] blocks = blockInfoList.toArray();
+        if (blocks != null) {
```

Comment:
why this array could be null?

Reply:
Actually not, but IDE always has an inspection note,  so I write this safe code to eliminate it.